executor/builder.go

```diff
@@ -778,12 +776,12 @@ func (b *executorBuilder) buildUnion(v *plan.Union) Executor {
 
 func (b *executorBuilder) buildUpdate(v *plan.Update) Executor {
 	tblID2table := make(map[int64]table.Table)
-	for id := range v.Schema().TblID2Handle {
+	for id := range v.SelectPlan.Schema().TblID2Handle {
 		tblID2table[id], _ = b.is.TableByID(id)
 	}
 	return &UpdateExec{
 		baseExecutor: newBaseExecutor(nil, b.ctx),
-		SelectExec:   b.build(v.Children()[0]),
+		SelectExec:   b.build(v.SelectPlan),
```

Comment:
after building, we should check `b.err`:
```go
selectExec := b.build(v.SelectPlan)
if b.err != nil {
    b.err = errors.Trace(b.err)
    return nil
}
```

Reply:
if b.err is not nil, it's ok to return any executor.

Reply:
It's better to discover error as soon as possible so that we can avoid useless executor building.